File: Zeze/Game/ProviderImplementWithOnline.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Zeze.Arch;
using Zeze.Builtin.Provider;
using Zeze.Net;
using Zeze.Transaction;

namespace Zeze.Game
{
    public class ProviderImplementWithOnline : Zeze.Arch.ProviderImplement
    {
        public Online Online { get; set; }

        protected override async Task<long> ProcessLinkBroken(Protocol _p)
        {
            var p = _p as LinkBroken;
            // 目前仅需设置online状态。
            if (false == string.IsNullOrEmpty(p.Argument.Context))
            {
                var roleId = long.Parse(p.Argument.Context);
                await Online.OnLinkBroken(roleId, ProviderService.GetLinkName(p.Sender), p.Argument.LinkSid);
            }
            return Procedure.Success;
        }
    }
}

File: ZezeJava/ZezeJava/src/main/java/Zeze/Game/ProviderImplementWithOnline.java
package Zeze.Game;

import Zeze.Arch.ProviderImplement;
import Zeze.Arch.ProviderService;
import Zeze.Builtin.Provider.LinkBroken;
import Zeze.Transaction.Procedure;

public class ProviderImplementWithOnline extends ProviderImplement {
	public Online online; // 需要外面初始化。App.Start.

	@Override
	protected long ProcessLinkBroken(LinkBroken p) throws Exception {
		// 目前仅需设置online状态。
		if (!p.Argument.getContext().isEmpty()) {
			var roleId = Long.parseLong(p.Argument.getContext());
			online.onLinkBroken(roleId, ProviderService.getLinkName(p.getSender()), p.Argument.getLinkSid());
		}
		return Procedure.Success;
	}
}
